cognite/client/utils/_json.py

```diff
@@ -1,13 +1,27 @@
 from __future__ import annotations
 
-import json
 import math
 import numbers
 from decimal import Decimal
 from types import MappingProxyType
 from typing import Any
 
-__all__ = ["dumps", "loads", "convert_to_float", "convert_nonfinite_float_to_str"]
+# Copied from requests.compat.py (except for type ignores):
+# json/simplejson module import resolution
+has_simplejson = False
+try:
+    import simplejson as json
+
+    has_simplejson = True
+except ImportError:
+    import json  # type: ignore [no-redef]
+
+if has_simplejson:
+    from simplejson import JSONDecodeError
+else:
+    from json import JSONDecodeError  # type: ignore [assignment]
```

Comment:
Let's add a comment about _why_ we're doing this instead, like you're great description in the PR.
```suggestion
# Users are seeing JSONDecodeError coming from a block that intercepts JSONDecodeError 
# (i.e. shouldn't be possible). It seems Databricks patches the built-in json library with simplejson, 
# and thus simplejson.JSONDecodeError != json.JSONDecodeError
try:
    import simplejson as json
    from simplejson import JSONDecodeError
except ImportError:
    import json  # type: ignore [no-redef]
    from json import JSONDecodeError  # type: ignore [assignment]
```